drivers/lora/sx126x.c

```diff
@@ -72,17 +73,47 @@ struct sx126x_data {
 	DioIrqHandler *radio_dio_irq;
 #if HAVE_GPIO_ANTENNA_ENABLE
 	struct device *antenna_enable;
+#endif
+#if HAVE_GPIO_TX_ENABLE
+	struct device *tx_enable;
+#endif
+#if HAVE_GPIO_RX_ENABLE
+	struct device *rx_enable;
 #endif
 	struct device *spi;
 	struct spi_config spi_cfg;
 #if HAVE_GPIO_CS
 	struct spi_cs_control spi_cs;
 #endif
+	RadioOperatingModes_t mode;
 } dev_data;
 
 
 void SX126xWaitOnBusy(void);
 
+static const char *sx126x_mode_name(RadioOperatingModes_t m)
+{
+	static const char *unknown_mode = "unknown";
+#define MODE(m) [MODE_##m] = #m
+	static const char *const names[] = {
+		MODE(SLEEP),
+		MODE(STDBY_RC),
+		MODE(STDBY_XOSC),
+		MODE(FS),
+		MODE(TX),
+		MODE(RX),
+		MODE(RX_DC),
+		MODE(CAD),
+	};
+#undef MODE
```

Comment:
What benefit are we getting by defining this inside local function? I mean with this `define & undef` tricks. Why can't you just declare this outside of `sx126x_mode_name` with fixed mode names?

Reply:
The macro is really just there to ensure that the names are consistent. I could put the table in at the file level, but keeping it in the function avoids polluting the name space and it should only be accessed using the function to avoid going out of bounds (in case there are new modes). Happy to move it if you think it makes the code more readable.

Reply:
I think we can just have it outside of this function as that makes the code readable.

Reply:
Done.